scripts/dev/upstream_merge/git_commands.py

```diff
@@ -0,0 +1,190 @@
+from shell_commands import *
+
+def git_clone(repo, directory=None, depth=None, capture_output=True):
+    """
+    Clone a Git repository.
+    :param repo: URL of the repository to clone.
+    :param directory: Directory to clone into (optional).
+    :param depth: Depth for shallow cloning (optional).
+    :param capture_output: Whether to capture the command output.
+    """
+    command = f"git clone {repo} {directory}" if directory else f"git clone {repo}"
+    if depth:
+        command += f" --depth {depth}"
+    return run_command(command, capture_output)
+
+def git_commit(message, amend=False, capture_output=True):
+    """
+    Commit changes to the repository.
+    :param message: Commit message.
+    :param amend: Whether to amend the previous commit.
+    :param capture_output: Whether to capture the command output.
+    """
+    if amend:
+        command = f"git commit --amend -m \"{message}\""
+    else:
+        command = f"git commit -m \"{message}\""
+    return run_command(command, capture_output)
+
+def get_current_commit(capture_output=True):
+    """
+    Get the current HEAD commit hash.
+    :param capture_output: Whether to capture the command output.
+    :return: The current commit hash.
+    """
+    return run_command("git rev-parse HEAD", capture_output)[1]
+
+def branch_exists(branch_name, capture_output=True):
+    """
+    Check if a branch exists locally.
+    :param branch_name: Name of the branch to check.
+    :param capture_output: Whether to capture the command output.
+    :return: True if the branch exists, False otherwise.
+    """
+    return run_command(f"git rev-parse --verify {branch_name}", capture_output)[0] == 0
```

Comment:
These two functions should be moved to the GitRepo.py file. We'll make the git_commands.py file contain the main git APIs. Things like these are what we specifically need for the upstream merge.